### PR TITLE
expose DD_CAPTURE_LAMBDA_PAYLOAD_MAX_DEPTH [SVLS-3853]

### PR DESCRIPTION
**Pull request by joeyzhao2018**
### What does this PR do?

1. Expose `DD_CAPTURE_LAMBDA_PAYLOAD_MAX_DEPTH` environment variable.  This is to achieve feature parity. Universal instrumentation already enabled the env: https://github.com/DataDog/datadog-agent/pull/19450
2. Add a section in Readme file to serve as a doc for the environment variables introduced in this repo.
3. small fix on the integration test's normalization for  package version data
### Types of Changes

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [x] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [x] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog
- [ ] This PR passes the integration tests (ask a Datadog member to run the tests)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,6 +16,21 @@ Follow the [installation instructions](https://docs.datadoghq.com/serverless/ins
...
+| DD_MIN_COLD_START_DURATION |  Sets the minimum duration (in milliseconds) for a module load event to be traced via Cold Start Tracing. Number. Defaults to `3`. |
+| DD_COLD_START_TRACE_SKIP_LIB | optionally skip creating Cold Start Spans for a comma-separated list of libraries. Useful to limit depth or skip known libraries. Default depends on runtime. |
+| DD_CAPTURE_LAMBDA_PAYLOAD | [Captures incoming and outgoing AWS Lambda payloads][1] in the Datadog APM spans for Lambda invocations. Defaults to `false`. |
+| DD_CAPTURE_LAMBDA_PAYLOAD_MAX_DEPTH | The captured AWS Lambda payloads will become tags of the `aws.lambda` span. This sets how deep it fathoms the JSON structure. When the max depth reached, the tag's value will be the stringified value of the deeper nested items. Defaults to `10`. <br> For example, with input payload as <pre>{<br>  "lv1" : {<br>    "lv2": {<br>      "lv3": "val"<br>    }<br>  }<br>}</pre> When set to `2`, the resulted tag's key is `function.request.lv1.lv2` and value `{\"lv3\": \"val\"}`. <br> When set to `0`, the the resulted tag's key is just `function.request` and value is `{\"lv1\":{\"lv2\":{\"lv3\": \"val\"}}}` |
```
Comment (joeyzhao2018) [inline README.md:31]: Need some help from docs team to improve the wording.
Screenshot 2023-11-09 at 7 59 40 AM

File: tests/test_tag_object.py [hunk 2]
```diff
@@ -30,6 +30,40 @@ def test_tag_object(self):
...
+                call(
+                    "function.request.level1.level2_list",
+                    "[None, True, 'nice', {'l3': 'v3'}]",
+                ),
+                call("function.request.level1.level2_bool", "True"),
+                call("function.request.level1.level2_int", "2"),
+            ],
+            True,
+        )
+        self.assertEqual(1, 1)
```
Comment (duncanista) [inline tests/test_tag_object.py:65]: What is this assert for?

File: tests/test_tag_object.py [hunk 3]
```diff
@@ -30,6 +30,40 @@ def test_tag_object(self):
...
+        payload = {
+            "hello": "world",
+            "level1": {
+                "level2_dict": {"level3": 3},
+                "level2_list": [None, True, "nice", {"l3": "v3"}],
+                "level2_bool": True,
+                "level2_int": 2,
+            },
+            "vals": [{"thingOne": 1}, {"thingTwo": 2}],
+        }
```
Comment (duncanista) [inline tests/test_tag_object.py:43]: Shouldn't we add a value with a deeper level to see if the it doesn't appear in the `function.request`?

File: tests/test_tag_object.py (hunk 2 above)
Comment (joeyzhao2018) [reply tests/test_tag_object.py]: sorry... i copied the previous test... now removed.

File: tests/test_tag_object.py (hunk 3 above)
Comment (joeyzhao2018) [reply tests/test_tag_object.py:42]: So the deeper-level items will be converted into one string (so they still appear but in a big chunk of string). For example, the "lv3", "level3" etc are already the "deeper level" of the setting here (2), and they would end up in the string value....

Or.... do you mean add an lv4 item?

Comment (duncanista) [reply tests/test_tag_object.py:42]: Ah I see, thank you so much, I was misunderstanding the feature!

**Review by duncanista — APPROVED** [review]
LGTM on the code side!

**Review by astuyve — APPROVED** [review]

File: datadog_lambda/wrapper.py [hunk 4]
```diff
@@ -72,10 +67,34 @@
...
+        )
+        return default_value
+
+
+dd_capture_lambda_payload_enabled = (
+    os.environ.get(DD_CAPTURE_LAMBDA_PAYLOAD, "false").lower() == "true"
+)
+
+if dd_capture_lambda_payload_enabled:
+    import datadog_lambda.tag_object as tag_object
```
Comment (astuyve) [inline datadog_lambda/wrapper.py:92]: nice lazy loading!!

File: README.md (hunk 1 above)
Comment (brett0000FF) [inline README.md:31]:
```suggestion
| DD_CAPTURE_LAMBDA_PAYLOAD_MAX_DEPTH | Determines the level of detail captured from AWS Lambda payloads, which are then assigned as tags for the `aws.lambda` span. It specifies the nesting depth of the JSON payload structure to process. Once the specified maximum depth is reached, the tag's value is set to the stringified value of any nested elements beyond this level. The default is `10`. <br> For example, given the input payload: <pre>{<br>  "lv1" : {<br>    "lv2": {<br>      "lv3": "val"<br>    }<br>  }<br>}</pre> If the depth is set to `2`, the resulting tag's key is set to `function.request.lv1.lv2` and the value is `{\"lv3\": \"val\"}`. <br> If the depth is set to `0`, the resulting tag's key is set to `function.request` and value is `{\"lv1\":{\"lv2\":{\"lv3\": \"val\"}}}` |
```
Here's a stab at improving the wording. Let me know if I misinterpreted anything.

Side note: I think you could make the table have 3 columns to separate out the default values into their own column.

**Review by brett0000FF — APPROVED** [review]

**Review by duncanista — APPROVED** [review]

**Pull request merged**